bangazonapi/views/profile.py

```diff
@@ -306,12 +306,27 @@ def favoritesellers(self, request):
                 }
             ]
         """
-        customer = Customer.objects.get(user=request.auth.user)
-        favorites = Favorite.objects.filter(customer=customer)
+        if request.method == 'GET':
+            customer = Customer.objects.get(user=request.auth.user)
+            favorites = Favorite.objects.filter(customer=customer)
 
-        serializer = FavoriteSerializer(
-            favorites, many=True, context={'request': request})
-        return Response(serializer.data)
+            serializer = FavoriteSerializer(
+                favorites, many=True, context={'request': request})
+            return Response(serializer.data)
+
+        if request.method == 'POST':
```

Comment:
Customers should only be able to 'favorite' a seller once. Before creating a new favorite object, you'll need to checks if the user has already 'favorited' that seller.